dataflow/flex-templates/streaming_beam/streaming_beam_test.py

```diff
@@ -0,0 +1,167 @@
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+import multiprocessing as mp
+import os
+import subprocess as sp
+import tempfile
+import time
+
+from google.cloud import bigquery
+from google.cloud import pubsub
+
+PROJECT = os.environ["GCLOUD_PROJECT"]
+DATASET = 'beam_samples'
+TABLE = 'streaming_beam_sql'
+TOPIC = 'messages'
+SUBSCRIPTION = 'messages'
+
+
+def create_topic_path():
+    publisher_client = pubsub.PublisherClient()
+    topic_path = publisher_client.topic_path(PROJECT, TOPIC)
+    try:
+        publisher_client.delete_topic(topic_path)
+    except Exception:
+        pass
+    response = publisher_client.create_topic(topic_path)
+    return response.name
+
+
+def create_subscription(topic_path):
+    subscriber = pubsub.SubscriberClient()
+    sub_path = subscriber.subscription_path(PROJECT, SUBSCRIPTION)
+    try:
+        subscriber.delete_subscription(sub_path)
+    except Exception:
+        pass
+    response = subscriber.create_subscription(sub_path, topic_path)
+    return response.name
+
+
+def _infinite_publish_job(topic_path):
+    publisher_client = pubsub.PublisherClient()
+    while True:
+        future = publisher_client.publish(
+            topic_path,
+            b'{"url": "https://beam.apache.org/", "review": "positive"}')
+        future.result()
+        time.sleep(10)
+
+
+def test_dataflow_flex_templates_pubsub_to_bigquery():
+    # Create BigQuery client and dataset.
+    bigquery_client = bigquery.Client(project=PROJECT)
+    dataset_id = '{}.{}'.format(PROJECT, DATASET)
+    dataset = bigquery.Dataset(dataset_id)
+    dataset = bigquery_client.create_dataset(dataset, exists_ok=True)
+
+    # Create Pubsub topic and subscription.
+    topic_path = create_topic_path()
+    sub_path = create_subscription(topic_path)
+
+    # Use one process to publish messages to a topic.
+    publish_process = mp.Process(target=lambda: _infinite_publish_job(topic_path))
+
+    # Use another process to run the streaming pipeline that should write one
+    # row to BigQuery every minute (according to the default window size).
+    pipeline_process = mp.Process(target=lambda: sp.call([
+        'python',
+        'streaming_beam.py',
+        '--project',
+        PROJECT,
+        '--runner',
+        'DirectRunner',
+        '--temp_location',
+        tempfile.mkdtemp(),
+        '--input_subscription',
+        sub_path,
+        '--output_table',
+        '{}:{}.{}'.format(PROJECT, DATASET, TABLE),
+    ]))
+
+    publish_process.start()
+    pipeline_process.start()
+
+    pipeline_process.join(timeout=90)
```

Comment:
Since we would be doing 1 second instead of 10 for each message, we can go lower than this. Maybe 5 or 10 seconds?

Reply:
I would say lets keep this to be 90 seconds. I have seen that with DirectRunner sometimes things  (like auto table creation) can take slightly longer and thus it can be indeterministic. Also, will have to modify and introduce a flag in the sample for the windowed duration interval if we were to reduce it by a lot.
Also, I dont think we need the sleep later so we can shave off some time there.

Reply:
Have introduced a window_interval param. Setting that to 5 seconds here and also keeping the timeout to 30 seconds.